File: 08-recursive-remainder/recursive_remainder.py
def recursive_remainder(a: int, b: int) -> int:
    # Handle negative numbers
    a = abs(a)
    b = abs(b)
    
    # Base case
    if a < b:
        return a
    
    # Recursive case
    return recursive_remainder(a - b, b)

def main():
    # Read input
    with open('input.txt', 'r') as f:
        a, b = map(int, f.read().strip().split())
    
    # Calculate remainder
    result = recursive_remainder(a, b)
    
    # Write output
    with open('output.txt', 'w') as f:
        f.write(str(result))

if __name__ == "__main__":
    main() 

File: 08-recursive-remainder/recursive_remainder.cpp
#include <iostream>
#include <fstream>
#include <cstdlib>

int recursiveRemainder(int a, int b) {
    // Handle negative numbers
    a = abs(a);
    b = abs(b);
    
    // Base case
    if (a < b) {
        return a;
    }
    
    // Recursive case
    return recursiveRemainder(a - b, b);
}

int main() {
    // Read input
    std::ifstream input("input.txt");
    int a, b;
    input >> a >> b;
    input.close();
    
    // Calculate remainder
    int result = recursiveRemainder(a, b);
    
    // Write output
    std::ofstream output("output.txt");
    output << result;
    output.close();
    
    return 0;
} 